data/models/auth-model.js

```diff
@@ -0,0 +1,35 @@
+const db = require('../../utilities/dbConfig');
+const bcrypt = require('bcryptjs');
+
+const checkUsername = async username => {
+  if (
+    await db('users')
+      .where({ username: username.toLowerCase() })
+      .first()
+  ) {
+    return 'taken';
+  } else {
+    return 'available';
+  }
+};
+
+const registerUser = async user => {
+  const password = bcrypt.hashSync(user.password, 8);
+  return db('users').insert({
+    username: user.username.toLowerCase(),
+    password,
+    department: user.department
```

Comment:
make it even more fun -- destructure all the properties off of the passed-in user and do the equivalent property-value syntax for all three properties